source/Pages/compute_on_spider.rst

```diff
@@ -305,7 +318,408 @@ combination that is not available you will receive the following error message:
+
+.. code-block:: bash
+
+   singularity exec --nv ubuntu.sif echo "hello world"
+
+.. WARNING::
+   The ``--nv`` flag is necessary to expose the GPUs on the host the to container.
+
+Here follows an example for running the container in batch mode with a shell script:
+
+.. code-block:: bash
+
+   #!/bin/bash
+
+   #SBATCH -p gpu_v100
+   #SBATCH -G v100:1
+   #SBATCH -e slurm-%j.out
+   #SBATCH -o slurm-%j.out
+
+   singularity exec --nv ubuntu.sif echo "hello world"
+
+The flags ``-e`` and ``-o`` instruct SLURM in which files to write respectively *stderr* and *stdout* of the job. In this case they are both sent to the same file, this is for comparison in the next step. If you now run this shell script on the ``ui-[01-02]`` nodes with ``bash script.sh``, it will result in:
+
+.. code-block:: bash
+
+   INFO:    Could not find any nv files on this host!
+   hello world
+
+as the UI nodes do not have access to GPUs and thus do not have an nv file to point the container to the required libraries. Running the script in batch mode with ``sbatch script.sh``, the ``-p`` flag is used, and the job ends up on a GPU node. The output becomes:
+
+.. code-block:: bash
+
+   hello world
+
+Of course, this ubuntu image does not have any of the tools needed to build GPU-native code or libraries that can run on the GPU. Refer to :ref:`this section <resources-singularity>` for more resources.
+
+.. tip::
+
+   While you do not get the warning about finding the nv file when using the ``--nv`` flag, you also have to specify which GPU to use, otherwise none are allocated to you! This can be done with the ``-G`` flag, as can be seen in the example shell script. 
+
+Now you are ready to build on top of a base container and run your code on a GPU!
+
+.. _accounting-gpu:
+
+Accounting of GPU usage
+=======================
+
+Currently the usage of GPU nodes is accounted for in GPU hours. This means that even though multiple cores are used simultaneously, one hour of use of a GPU node is billed as 1 GPU-hour. Unlike CPUs where one hour of multi-core usage is billed as multiple CPU hours, depending on the number of cores.
+
+.. _singularity-building:
+
+================================
+Building and running a singularity container
+================================
+
+In this section we show how to build a singularity container use it to run code in its environment. There is extensive documentation from singularity itself `here <https://docs.sylabs.io/guides/latest/user-guide/index.html>`_. 
+
+The steps in this section are done on GPU nodes, to ensure availability of the drivers, which may be needed in some compilation steps.
+
+Building directly from dockerhub
+================================
+
+There are multiple ways to build a container. To build directly from docker hub, for example the latest version of tensorflow, one can invoke:
+
+.. code-block:: bash
+   
+   singularity build --nv tf_latest.sif docker://tensorflow/tensorflow:latest
```

Comment:
should this not contain the --fakeroot option, or since this is just a format conversion its not necessary (if the latter then perhaps very briefly state this and perhaps consider explaining the logical difference between image conversion vs. building from a recipe file)

Reply:
Fakeroot is only necessary when raising to root during the building process (so with a definitions file) and calling commands like apt, yum, etc. But not for pip, as it does not require root.